gcp_variant_transforms/transforms/partition_variants_test.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for partition_variants module."""
+
+from __future__ import absolute_import
+
+import unittest
+
+from apache_beam import Partition
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+from apache_beam.transforms import Create
+
+from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.transforms import partition_variants
+from gcp_variant_transforms.libs import variant_partition
+
+
+class PartitionVariantsTest(unittest.TestCase):
+  """Test cases for the ``PartitionVariants`` transform."""
+
+  def _get_standard_variants(self):
+    # Valid variants reference_name.strip().lower() will successfully matched to
+    #   re.compile(r'^(chr)?([0-9][0-9]?)$')
+    variants = []
+    # Partition 0
+    variants.append(vcfio.Variant(reference_name='chr1', start=0))
+    variants.append(vcfio.Variant(reference_name='CHR1', start=0))
+    # Partition 1
+    variants.append(vcfio.Variant(reference_name='cHr2', start=0))
+    variants.append(vcfio.Variant(reference_name='chR2', start=0))
+    # Partition 2
+    variants.append(vcfio.Variant(reference_name='chr3', start=0))
+    variants.append(vcfio.Variant(reference_name='chr03', start=0))
+    # Partition 5
+    variants.append(vcfio.Variant(reference_name='6', start=0))
+    variants.append(vcfio.Variant(reference_name='06', start=0))
+    variants.append(vcfio.Variant(reference_name='chr06', start=0))
+    # Partition 9
+    variants.append(vcfio.Variant(reference_name='chr10', start=0))
+    variants.append(vcfio.Variant(reference_name='  chr10   ', start=0))
+    # Partition 21
+    variants.append(vcfio.Variant(reference_name='chr22', start=0))
+    variants.append(vcfio.Variant(reference_name='chr22    ', start=0))
+    return variants, variants[0:2], variants[2:4], variants[4:6],\
```

Comment:
Instead of this, I think it's much more readable to return a dict where the key is the partition number (int) and the value is a list of variants that should map to that partition. You can just return that dict both here and in non-standard case.
In your test, you would essentially pass the dict values as inputs to the pipeline, and then loop through [0, get_num_partitions) and if the key is in the dict, make sure that it matches the value, if not make sure it's empty. WDYT?
P.S. Is checking the 'label' important? If so, I suppose you can have a tuple of partition number and label as the key to your dict.